### PR TITLE
Fix/sonar multiple servers

### LINKED ISSUE #2971: SonarQube collector does not support multiple servers with multiple accounts

### LINKED ISSUE #2972: SonarQube properties and UI config conflict with each other
I have multiple sonar qube instances (for historical reasons). If I make a modification to the configuration through the UI the records get reordered in the database. However in the properties file there are version numbers and REST query URLs that are hardcoded to a certain array index...

Obviously this causes problems. Should be fetching the version from the sonar instance. and the metric string can just be part of the client class.

### PR DESCRIPTION
**Pull request by fidelcoria-aa**
Fix #2972 by moving metric strings and version numbers out of properties file and into collector
  the Config from admin page overrides any config from properties file
Fix #2971 by supporting multiple usernames and passwords for multiple sonar instances

### CODE REVIEW

File: collectors/build/sonar/src/main/java/com/capitalone/dashboard/collector/SonarCollectorTask.java [hunk 1]
```diff
@@ -131,25 +158,23 @@ public void collect(SonarCollector collector) {
...
-                    for (CollectorItem ci : itemList) {
-                        if (ci != null && ci.getCollectorId().equals(collector.getId())) {
-                            uniqueIDs.add(ci.getId());
-                        }
-                    }
-                }
-            }
-        }
+        // extract unique collector item IDs from components
+        // (in this context collector_items are sonar projects)
```
Comment (rvema) [inline collectors/build/sonar/src/main/java/com/capitalone/dashboard/collector/SonarCollectorTask.java]: +1

**Comment by rvema** [comment]
@fidelcoria-aa When I try to run the collector locally, I get errors especially around metrics section. I am moving this repository to a new location this weekend. Can we setup a call to discuss this PR?

**Comment by fidelcoria-aa** [comment]
We are using Sonar versions 6.0 and 6.7. I don't see any metrics errors. What version are you testing with?
@rvema, send me an email with the call details.

**Comment by rvema** [comment]
@fidelcoria-aa I will setup a call next week and send you the details of the meeting. Meanwhile we have moved the sonar collector to its own repository can you please close this PR and create it against the new repo? Here is the link to the same - https://github.com/Hygieia/hygieia-codequality-sonar-collector

**Comment by fidelcoria-aa** [comment]
Yes, sir!

**Comment by fidelcoria-aa** [comment]
I have split up this PR into 5 smaller ones. I hope this helps
Hygieia/hygieia-codequality-sonar-collector#5
Hygieia/hygieia-codequality-sonar-collector#6
Hygieia/hygieia-codequality-sonar-collector#7
Hygieia/hygieia-codequality-sonar-collector#8
Hygieia/hygieia-codequality-sonar-collector#9

**Pull request closed without merging**